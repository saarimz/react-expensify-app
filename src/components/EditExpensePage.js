import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { editExpense, startRemoveExpense } from '../actions/expenses'

export class EditExpensePage extends React.Component {
    constructor(props){
        super(props);
        this.onRemove= this.onRemove.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
    }
    onRemove(){
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    }
    onSubmit(expense){
        this.props.editExpense(this.props.expense.id,expense);
        this.props.history.push('/');
    }
    render(){
        return (
            <div>
                <ExpenseForm 
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                    action="Edit Expense"
                />
                <button onClick={this.onRemove}>Remove</button>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch, props) => {
    return ({
        editExpense: (id,expense) => {
            dispatch(editExpense(id,expense));
        },
        startRemoveExpense: (data) => {
            dispatch(startRemoveExpense(data));
        }
    });
};

const mapStateToProps = (state,props) => {
    return {
        expense: state.expenses.find((expense) => (expense.id == props.match.params.id))
    }
};

export default connect(mapStateToProps,mapDispatchToProps)(EditExpensePage);