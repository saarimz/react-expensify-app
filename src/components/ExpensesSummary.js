import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({expensesTotal, expensesCount}) => (
    <div>
        <p>Viewing {expensesCount} {expensesCount == 1 ? 'expense' : 'expenses'} totalling {numeral(expensesTotal / 100).format('$0,0.00')}</p>
    </div>
);

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses,state.filters);
    return {
        expensesTotal: getExpensesTotal(visibleExpenses),
        expensesCount: visibleExpenses.length
    }
};

export default connect(mapStateToProps)(ExpensesSummary);