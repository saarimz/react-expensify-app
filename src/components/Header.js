import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <ul>
            <li><NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink></li>
            <li><NavLink to="/create" activeClassName="is-active" exact={true}>New Expense</NavLink></li>
        </ul>
    </header>
);

export default Header;