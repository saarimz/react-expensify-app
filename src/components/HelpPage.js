import React from 'react';

const HelpPage = () => (
    <div>
        <h1>This is the help component</h1>
    </div>
);

export default HelpPage;
