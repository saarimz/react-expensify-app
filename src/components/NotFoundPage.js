import React from 'react';
import {Link} from 'react-router-dom';

const NotFoundPage = () => (
    <div>
        <h1>404! Ooops, not found Go to: <Link to="/">Home</Link></h1>
    </div>
);

export default NotFoundPage;