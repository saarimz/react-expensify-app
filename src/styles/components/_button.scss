.bttn {
    background-color: $blue;
    color: $light-grey;
    padding: $s-size $m-size;
    margin: 0 auto;
    border: none;
    border-radius: $xs-size;
    font-weight: bolder;
    cursor: pointer;
    font-size: $font-size-large;
    display: inline-block;
    text-decoration: none;
    line-height: 1;
}

.bttn--link {
    border: none;
    margin: 0;
    background: none;
}

.bttn--secondary {
    background-color: grey;
    color: $white;
}