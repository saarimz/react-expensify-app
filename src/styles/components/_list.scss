.list-header {
    background: $light-grey;
    border: 1px solid darken($light-grey,10%);
    color: $grey;
    display: flex;
    justify-content: space-between;
    padding: $s-size $m-size;
}

.list-item {
    border: 1px solid darken($light-grey,10%);
    border-top: none;
    display: flex;
    flex-direction: column;
    text-decoration: none;
    color: $dark-grey;
    padding: $s-size;
    background: $white;
    transition: background 0.3s ease;
    &:hover {
        background: $light-grey;
    }

    @media screen and (min-width: $desktop-breakpoint) {
        flex-direction: row;
        justify-content: space-between;
        padding: $m-size;
        align-items: center;
    }
}

.list-item__title {
    margin: 0;
    word-break: break-all;
}

.list-item__subtitle {
    color: $grey;
    font-size: $font-size-small;
}

.list-item__data {
    margin: $s-size 0 0 0;
    @media screen and (min-width: $desktop-breakpoint) {
        margin: 0;
        padding-left: $s-size;
    }
}

.list-item--message {
    align-items: center;
    justify-content: center;
    padding: $m-size;
    color: $grey;
    &:hover {
        background: none;
    }
}

.list-body {
    margin-bottom: $m-size;
    @media screen and (min-width: $desktop-breakpoint) {
        margin-bottom: $l-size;
    }
}