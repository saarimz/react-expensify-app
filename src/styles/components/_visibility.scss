.show-for-mobile {
    display: block;
    @media screen and (min-width: $desktop-breakpoint) {
        display: none;
    }
}

.show-for-desktop {
    display: none;
    @media screen and (min-width: $desktop-breakpoint + 0.01rem) {
        display: block;
    }
}