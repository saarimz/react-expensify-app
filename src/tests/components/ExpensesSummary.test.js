import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render component with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={420}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render component correctly with 2 expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={2} expensesTotal={5000.23}/>);
    expect(wrapper).toMatchSnapshot();
});