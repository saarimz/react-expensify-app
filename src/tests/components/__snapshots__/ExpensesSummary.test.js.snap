// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component correctly with 2 expenses 1`] = `
<div>
  <p>
    Viewing 
    2
     
    expenses
     totalling 
    $50.00
  </p>
</div>
`;

exports[`should render component with 1 expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $4.20
  </p>
</div>
`;
